installer/pkg/nextgen/asset/installconfig.go

```diff
@@ -0,0 +1,72 @@
+package asset
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/pborman/uuid"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/installer/pkg/nextgen"
+)
+
+type InstallConfig struct {
+	assetGraph *Graph
+}
+
+var _ nextgen.Asset = (*InstallConfig)(nil)
+
+func (a *InstallConfig) Dependencies() []nextgen.Asset {
+	return []nextgen.Asset{
+		a.assetGraph.EmailAddress,
+		a.assetGraph.Password,
+		a.assetGraph.BaseDomain,
+		a.assetGraph.ClusterName,
+		a.assetGraph.License,
+		a.assetGraph.PullSecret,
+		a.assetGraph.Platform,
+	}
+}
+
+func (a *InstallConfig) Generate(dependencies []*nextgen.AssetState) (*nextgen.AssetState, error) {
+	//emailAddress := string(dependencies[0].Contents[0].Data)
+	//password := string(dependencies[1].Contents[0].Data)
+	//baseDomain := string(dependencies[2].Contents[0].Data)
+	clusterName := string(dependencies[3].Contents[0].Data)
+	//license := string(dependencies[4].Contents[0].Data)
+	//pullSecret := string(dependencies[5].Contents[0].Data)
+	platform := string(dependencies[6].Contents[0].Data)
+
+	installConfig := nextgen.InstallConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterName,
+		},
+		ClusterID: uuid.NewUUID(),
+	}
+
+	switch platform {
+	case AWSPlatformType:
+		installConfig.AWS = &nextgen.AWSPlatform{}
+	case LibvirtPlatformType:
+		installConfig.Libvirt = &nextgen.LibvirtPlatform{}
+	default:
+		return nil, fmt.Errorf("unknown platform type %q", platform)
+	}
+
+	json, err := jsoniter.ConfigFastest.Marshal(installConfig)
```

Comment:
not sure why specfically marshal to json then, marshal to yaml is required. "github.com/ghodss/yaml".Marshal() already does it?